tests/c/conformance_test.c

```diff
@@ -2114,7 +2114,8 @@ static int test_bind_exec_direct_with_stmt(SQLHANDLE hconn, SQLHANDLE hstmt)
   CALL_SQLSetStmtAttr(hstmt, SQL_ATTR_PARAM_BIND_TYPE, SQL_PARAM_BIND_BY_COLUMN, 0);
 
   // Specify the number of elements in each parameter array.
-  CALL_SQLSetStmtAttr(hstmt, SQL_ATTR_PARAMSET_SIZE, (SQLPOINTER)(uintptr_t)nr_paramset_size, 0);
+  // CALL_SQLSetStmtAttr(hstmt, SQL_ATTR_PARAMSET_SIZE, (SQLPOINTER)(uintptr_t)nr_paramset_size, 0);
+  CALL_SQLSetStmtAttr(hstmt, SQL_ATTR_PARAMSET_SIZE, (SQLPOINTER)(uintptr_t)ARRAY_SIZE, 0);
```

Comment:
It is true that the code here should not be changed this way, but should keep the nr_paramset_size variable consistent, but I still have some confusion: Why initialize array values of ARRAY_SIZE(10) length instead of using only nr_paramset_size(4) values from the initial array? Can nr_paramset_size and ARRAY_SIZE be combined into one variable?

```
  const param_t params[] = {
    {SQL_PARAM_INPUT,  SQL_C_SBIGINT,  SQL_TYPE_TIMESTAMP,  23,       3,          ts_arr,           0,   ts_ind},
    {SQL_PARAM_INPUT,  SQL_C_CHAR,     SQL_VARCHAR,         20,       0,          varchar_arr,      100, varchar_ind},
    {SQL_PARAM_INPUT,  SQL_C_CHAR,     SQL_WVARCHAR,        20,       0,          nchar_arr,        100, nchar_ind},
    {SQL_PARAM_INPUT,  SQL_C_SBIGINT,  SQL_BIGINT,          8,        0,          i64_arr,          0,   i64_ind},
  };

  SQLULEN nr_paramset_size = 4;

  for (int i=0; i<ARRAY_SIZE; ++i) {
    ts_arr[i] = 1662861448751 + i;
    snprintf(varchar_arr[i], 100, "a人%d", i);
    varchar_ind[i] = SQL_NTS;
    snprintf(nchar_arr[i], 100, "b民%d", i);
    nchar_ind[i] = SQL_NTS;
    i64_arr[i] = 54321 + i;
  }
```

And I suggest that the value of nr_paramset_size and the size of params be numerically separated at the end to avoid confusion, just like the sample of Microsoft ODBC website: 
https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/binding-arrays-of-parameters?view=sql-server-ver16